### ISSUE TITLE
Create a comprehensive release test setup

### ISSUE THREAD

**Issue by wagerlabs** [aeternity/epoch#192]
We need to 

- catch issues like `badarith` which we missed twice in a row!
- not slow down the day-to-day regular Travis job stream, i.e. hook this up to the release build only.

**Comment by lucafavatella** [comment]
I understand the manual test approach is:
* Start release `bin/epoch start`;
* Ensure release is started i.e. ensure `bin/epoch ping` succeeds;
* Query top `curl http://127.0.0.1:3013/v1/top` (the port number is the default in the `sys.config`). Observe increasing height;
* Query balance `curl http://127.0.0.1:3013/v1/account/balance`. Observe increasing balance.

**Comment by lucafavatella** [comment]
> not slow down the day-to-day regular Travis job stream, i.e. hook this up to the release build only.

The job that is responsible for building the package and uploading the package to GitHub is called [`package`](https://github.com/aeternity/epoch/blob/a8f6eb2ad30a6adfd632c4a8f67a1d3e982c1469/.travis.yml), and runs (as of master) on 3 types on machine: Ubuntu, recent OSX/Xcode and old OSX/Xcode.

The 3 jobs (i.e. `package` on the 3 types of machine - hence 3 jobs) run alongside all other jobs (i.e. branch push, tag push, pull request open, pull request merge) by building the package. The jobs upload the package [only when the trigger is a tag](https://github.com/aeternity/epoch/blob/a8f6eb2ad30a6adfd632c4a8f67a1d3e982c1469/.travis.yml#L28).

**Comment by velzevur** [comment]
I think we can easily add to the manual approach a download of the whole chain. This would give us even more certainty that everything is fine.
We could easily do it with swagger generated python client

**Comment by lucafavatella** [comment]
> catch issues like badarith which we missed twice in a row!

I did not manage to reproduce `badarith` on a8f6eb2:
* For Ubuntu, I used package built by Travis and uploaded on GitHub, I tested it on VirtualBox (`vagrant init ubuntu/trusty64; vagrant up; vagrant ssh` then see content of `/vagrant`). No `badarith` error. Mining first block took a few minutes, mining second block is taking tens of minutes;
* For OSX, I did not use package. I cleaned my working copy, then I issued `make prod-build` and `make prod-start`. No `badarith` error. Mining first block took a few minutes, mining second block took tens of minutes, mining third block took a few minutes.
